cli/lsp/logging.rs

```diff
@@ -1,13 +1,65 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
 
+use deno_core::parking_lot::Mutex;
+use deno_runtime::tokio_util::create_basic_runtime;
+use std::fs;
+use std::io::prelude::*;
+use std::path::PathBuf;
 use std::sync::atomic::AtomicBool;
 use std::sync::atomic::AtomicUsize;
 use std::sync::atomic::Ordering;
+use std::thread;
 
 static LSP_DEBUG_FLAG: AtomicBool = AtomicBool::new(false);
 static LSP_LOG_LEVEL: AtomicUsize = AtomicUsize::new(log::Level::Info as usize);
 static LSP_WARN_LEVEL: AtomicUsize =
   AtomicUsize::new(log::Level::Warn as usize);
+static LOG_FILE: Mutex<Option<LogFile>> = Mutex::new(None);
+
+pub struct LogFile {
+  path: PathBuf,
+  buffer: String,
+}
+
+impl LogFile {
+  pub fn log(&mut self, s: &str) {
+    self.buffer.push_str(s);
+    self.buffer.push('\n');
+  }
+
+  fn commit(&mut self) {
```

Comment:
This will block the lock while being written. Perhaps instead we could take the data out of the lock in order to free it immediately, then append file.